src/screen/ProductPage.tsx

```diff
@@ -15,47 +15,66 @@ type ProductPageProps = {
   ProductClick: () => void;
 };
 
+type ProductForm = Omit<Product, "id"> & { price?: number };
+
 type InputType = "add" | "edit";
 
 export function ProductPage(props: ProductPageProps) {
-  const [inputProduct, setInputProduct] = useState<Product>({} as Product);
+  const [inputProduct, setInputProduct] = useState<ProductForm>({
+    title: "",
+    categoryId: 1,
+    price: 0,
+  });
+  // const [product, setProduct] = useState<Product | null>(null); // contoh
   const [inputType, setInputType] = useState<InputType>("add");
-  const [inputCategoryProduct, setInputCategoryProduct] = useState<number>();
-  const { products, reFetchProduct, isLoadingFetchProduct } = useFetchProduct();
-  const [editProduct, setEditProduct] = useState<number>(
-    products ? products[products.length - 1].id + 1 : 1
-  );
-  const { categories, reFetch, isLoadingFetch } = useFetchCategories();
-  const { isLoadingCreate, submit } = useCreateProduct();
-  const { delProduct, isLoadingDelete } = useDeleteProduct();
-  const { isLoadingEdit, updateProduct } = useEditProduct();
+  const [idEditProduct, setIdEditProduct] = useState<number>(0);
```

Comment:
Ini bisa dibuat null by default

```suggestion
  const [idEditProduct, setIdEditProduct] = useState<number | null>(null);
```

Reply:
already updated mas